File: OOAD_HW1/Q4/zooKeeper.java
package Q4;
import java.util.LinkedList;
import java.util.List;

/* Zookeeper Class
 * Takes in a zoo object as as parameter, iterates through the animals in the zoo, and
 * sends a message to each instance as to what method (makeNoise(), eat(), etc.) to call.
 */
public class zooKeeper {

	private zoo Z;
	List<animal> animals;	// This will come from the zoo object 'Z'
	
	public zooKeeper() {
		//Keep the zoo empty? Don't use the default.
	}
	
	// Set the zoo private member to a zoo object. The zookeeper manages this object.
	public zooKeeper(zoo Z){
		this.Z = Z;
		this.animals = Z.getAnimals();	// List of the animals at the zoo.
	}
	
	/* Zookeeper Actions
	 * iterates over the list, gets an animal from the zoo list, and tells it what to do.
	 * That instance of that animal object will perform the action.
	 */
	public void wakeUp() {
		System.out.println("The Zookeeper wakes up the animals of the zoo.");
		for (int i = 0; i < this.animals.size(); i++) {
			this.animals.get(i).wakeUp();
		}
	}
	
	public void rollCall() {
		System.out.println("The Zookeeper does a roll call of all the animals.");
		for (int i = 0; i < this.animals.size(); i++) {
			this.animals.get(i).makeNoise();
		}
	}
	public void feed() {
		System.out.println("The Zookeeper feeds the animals of the zoo.");
		for (int i = 0; i < this.animals.size(); i++) {
			this.animals.get(i).eat();
		}
	}
	public void exercise() {
		System.out.println("The Zookeeper exercises the animals of the zoo.");
		for (int i = 0; i < this.animals.size(); i++) {
			this.animals.get(i).roam();
		}
		
	}
	public void close() {
		System.out.println("The Zookeeper starts to close the zoo. The animals are going to sleep.");
		for (int i = 0; i < this.animals.size(); i++) {
			this.animals.get(i).sleep();
		}
	}
}

File: OOAD_HW2/Part B/Zookeeper.cpp
#include <iostream>
#include <string>
#include <cstdlib>
#include <iomanip>
#include <vector>

using namespace std;

class ZooKeeper
{
    private:
        Zoo z;
        vector<Animal> animals;

    public:
        ZooKeeper(Zoo zoo);
        void wakeUp(void);
        void rollCall(void);
        void feed(void);
        void exercise(void);
        void close(void);
};

ZooKeeper::ZooKeeper(Zoo zoo)
{
    // Set the zoo private member to a zoo object. The zookeeper manages this object.
    z = zoo;
    animals = z.getAnimals();
}

/* Zookeeper Actions
 * iterates over the list, gets an animal from the zoo list, and tells it what to do.
 * That instance of that animal object will perform the action.
 */
void ZooKeeper::wakeUp(void)
{
    cout << "The Zookeeper wakes up the animals of the zoo." << endl;

    for(int i = 0; i < animals.size(); i++)
    {
        animals[i].wakeUp();
    }
}

void ZooKeeper::rollCall(void)
{
    cout << "The Zookeeper does a roll call of all the animals." << endl;

    for(int i = 0; i < animals.size(); i++)
    {
        animals[i].makeNoise();
    }
}

void ZooKeeper::feed(void)
{
    cout << "The Zookeeper feeds the animals of the zoo." << endl;

    for(int i = 0; i < animals.size(); i++)
    {
        animals[i].eat();
    }
}

void ZooKeeper::exercise(void)
{
    cout << "The Zookeeper exercises the animals of the zoo." << endl;

    for(int i = 0; i < animals.size(); i++)
    {
        animals[i].roam();
    }
}

void ZooKeeper::close(void)
{
    cout << "The Zookeeper starts to close the zoo, the animals are going to sleep." << endl;

    for(int i = 0; i < animals.size(); i++)
    {
        animals[i].sleep();
    }
}
